pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/LeaderElectionImpl.java

```diff
@@ -102,10 +105,13 @@ private synchronized CompletableFuture<LeaderElectionState> elect() {
             } else {
                 return tryToBecomeLeader();
             }
-        }).thenCompose(leaderElectionState ->
-                // make sure that the cache contains the current leader
-                // so that getLeaderValueIfPresent works on all brokers
-                cache.get(path).thenApply(__ -> leaderElectionState));
+        }).thenComposeAsync(leaderElectionState -> {
+            // make sure that the cache contains the current leader
+            // so that getLeaderValueIfPresent works on all brokers
+            cache.refresh(path);
```

Comment:
is this a blocking operation ?

Reply:
it is. That's why I changed it to be executed within `thenComposeAsync`.